--- a/net/minecraft/world/entity/npc/villager/Villager.java
+++ b/net/minecraft/world/entity/npc/villager/Villager.java
@@ -254,6 +_,8 @@
     }
     // Paper end - EAR 2
 
+    public long nextGolemPanic = -1; // Crisp - DAB
+    private int behaviorTick = 0; // Crisp - DAB
     @Override
     protected void customServerAiStep(final ServerLevel level) {
         // Paper start - EAR 2
@@ -263,7 +_,11 @@
         // Paper end - EAR 2
         ProfilerFiller profiler = Profiler.get();
         profiler.push("villagerBrain");
-        if (!inactive) this.getBrain().tick(level, this); // Paper - EAR 2
+        // Crisp start - DAB
+        if (!inactive && this.behaviorTick++ % this.activatedPriority == 0) {
+            this.getBrain().tick(level, this); // Paper - EAR 2
+        }
+        // Crisp end - DAB
         profiler.pop();
         if (this.assignProfessionWhenSpawned) {
             this.assignProfessionWhenSpawned = false;
