--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -251,6 +_,7 @@
         this.server.spark.registerCommandBeforePlugins(this.server); // Paper - spark
         com.destroystokyo.paper.Metrics.PaperMetrics.startMetrics(); // Paper - start metrics
         com.destroystokyo.paper.VersionHistoryManager.INSTANCE.getClass(); // Paper - load version history now
+        io.github.madebywaffle.crisp.CrispConfig.load(); // Crisp - config
 
         // this.worldData.setGameType(properties.gameMode.get()); // CraftBukkit - moved to world loading
         LOGGER.info("Default game type: {}", properties.gameMode.get());
